src/types.ts

```diff
@@ -106,6 +107,7 @@ export interface ContractCallOptions {
   gas?: BigNumber | number;
   gasPrice?: BigNumber | number;
   nonce?: number;
+  promise?: boolean;
```

Comment:
`promise` isn't a very specific name. The calls will still return promises regardless so it's kinda confusing. How about:

```suggestion
waitForConfirmation?: boolean;
```

or something